apps/prairielearn/elements/pl-external-grader-results/pl-external-grader-results.py

```diff
@@ -39,129 +41,140 @@ def ansi_to_html(output):
-                        if not tests_missing_points:
-                            test["points"] = round_value(results_test.get("points"))
-                            test["max_points"] = round_value(
-                                results_test.get("max_points")
-                            )
-
-                            if test["max_points"] != test["points"]:
-                                test["results_color"] = "#E70000"
-                                test["results_icon"] = "fa-times"
-                            # Don't consider points for test cases that are 0/0
-                            # to be correct. We compare with a string because
-                            # `round_value()` returns a string.
-                            elif test["max_points"] == "0":
-                                test["results_color"] = "#757575"
-                                test["results_icon"] = "fa-circle-info"
-                            else:
-                                test["results_color"] = "#008900"
-                                test["results_icon"] = "fa-check"
-
-                        test["images"] = results_test.get("images", [])
-                        test["has_images"] = len(test["images"]) > 0
-                        tests.append(test)
-
-                    html_params["tests"] = tests
-
-        with open("pl-external-grader-results.mustache", "r", encoding="utf-8") as f:
-            html = chevron.render(f, html_params).strip()
-    else:
-        html = ""
-
-    return html
+def round_value(value: float, digits: int = 2) -> str:
+    """
+    Round the given value to the specified precision; default is 2 digits.
+
+    Remove trailing 0s and '.'s, e.g., convert "1.00" to "1".
+    """
+    return f"{value:.{digits}f}".rstrip("0").rstrip(".")
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    # Early-exit if not the submission panel
+    if data["panel"] != "submission":
+        return ""
+
+    feedback = data["feedback"]
+    results = feedback.get("results", None)
+
+    format_errors = data.get("format_errors", {})
+    grader_format_errors = format_errors.get("_external_grader", [])
+
+    grading_succeeded = bool(feedback.get("succeeded", None))
+
+    html_params = {
+        "submission": True,
+        "graded": bool(feedback),
+        "gradable": results["gradable"]
+        if (results is not None and "gradable" in results)
+        else True,
+        "grading_succeeded": grading_succeeded,
+        "format_errors": grader_format_errors,
+        "has_format_errors": len(grader_format_errors) > 0,
+        "uuid": pl.get_uuid(),
+    }
+
+    if grading_succeeded and results:
+        html_params["succeeded"] = bool(results.get("succeeded", None))
+        html_params["score"] = round_value(results.get("score", 0) * 100)
+        html_params["achieved_max_points"] = results.get("score", 0) >= 1.0
+        html_params["results_color"] = (
+            PLColor("correct_green")
```

Comment:
Does this automatically translate the `PLColor` instance to a string (which contains the hex value)? Not opposed to doing it this way, just curious.

Reply:
It's not a hex string, but it does get stringified into an rgba string. I'll try to copy an example here.